plugin/plugin_test.go

```diff
@@ -0,0 +1,385 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package plugin
+
+import (
+	"testing"
+
+	"github.com/containerd/containerd/services"
+)
+
+func RegisterClear() {
+	register.Lock()
+	defer register.Unlock()
+	register.r = nil
+}
+
+func MockPluginFilter(*Registration) bool {
+	return false
+}
+
+var tasksServiceRequires = []Type{
+	RuntimePlugin,
+	RuntimePluginV2,
+	MetadataPlugin,
+	TaskMonitorPlugin,
+}
+
+// TestContainerdPlugin tests the logic of Graph, use the containerd's plugin
+func TestContainerdPlugin(t *testing.T) {
```

Comment:
Why is this test needed? If the purpose is to test Graph, isn't TestPluginGraph just enough?

Reply:
TestContainerdPlugin tries to simulate the use of containerd as much as possible, Graph is the constructed test case. I think it is clearer to separate them.